### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by SeanFeldman**
- object initializers
- redundant type specifiers
- unnecessary public modifiers
- unnecessary null initialization for objects
- unnecessary parentheses
- unnecessary array type specifier
- unused variables
- nameof instead of hard-coded names
- unused namespace usings
- unneeded cast for enum values
- unnecessary boxing for nunit
- prefer async version of methods over sync

image

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs [hunk 1]
```diff
@@ -16,26 +16,26 @@ namespace Microsoft.Azure.ServiceBus.Amqp
...
-        public const string PublisherName = "x-opt-publisher";
-        public const string PartitionKeyName = "x-opt-partition-key";
-        public const string PartitionIdName = "x-opt-partition-id";
-        public const string PrefilteredMessageHeadersName = "x-opt-prefiltered-headers";
-        public const string PrefilteredMessagePropertiesName = "x-opt-prefiltered-properties";
-        public const string DeadLetterSourceName = "x-opt-deadletter-source";
-        public const string TimeSpanName = AmqpConstants.Vendor + ":timespan";
-        public const string UriName = AmqpConstants.Vendor + ":uri";
-        public const string DateTimeOffsetName = AmqpConstants.Vendor + ":datetime-offset";
+        private const string EnqueuedTimeUtcName = "x-opt-enqueued-time";
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs]: >private [](start = 8, length = 7)

nit: Remove explicit private modifier.. We have typically tried to keep this practice through our solution.

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs [hunk 2]
```diff
@@ -139,7 +139,7 @@ internal AmqpMessageSender(string entityName, MessagingEntityType entityType, Se
                     null);
             request.Map[ManagementConstants.Properties.SequenceNumbers] = new[] { sequenceNumber };
 
-            var response = await this.ExecuteRequestResponseAsync(request);
+            await this.ExecuteRequestResponseAsync(request);
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs]: >await this.ExecuteRequestResponseAsync(request); [](start = 12, length = 48)

we should handle a response that is not OK and throw exception here.

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 3]
```diff
@@ -81,10 +81,7 @@ protected override Task OnCloseAsync()
                 amqpRequestMessage.Map[ManagementConstants.Properties.RuleDescription] =
                     AmqpMessageConverter.GetRuleDescriptionMap(description);
 
-                AmqpResponseMessage response =
-                    await
-                        ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage)
-                            .ConfigureAwait(false);
+                await ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage).ConfigureAwait(false);
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: >await [](start = 16, length = 5)

Same here. we should handle Not OK case and throw with the correct exception here

File: src/Microsoft.Azure.ServiceBus/Primitives/MessagingEntityType.cs [hunk 4]
```diff
@@ -10,6 +10,6 @@ public enum MessagingEntityType
         Subscriber = 2,
         Filter = 3,
         Namespace = 4,
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Primitives/MessagingEntityType.cs]: >Namespace = 4, [](start = 8, length = 14)

could remove this

**Comment by vinaysurya** [comment]
Thanks Sean for doing this. Have some minor comments, could you please take a look and recreate PR

**Comment by jtaubensee** [comment]
Recreate? You just mean push a new commit, right?

**Comment by vinaysurya** [comment]
Yes

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs]: 👍 
BTW, `LockTokenName, `PrefilteredMessageHeadersName`, and `PrefilteredMessagePropertiesName` are never used...

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs (hunk 2 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs]: left a `TODO` comment

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs (hunk 3 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: same TODO

File: src/Microsoft.Azure.ServiceBus/Primitives/MessagingEntityType.cs (hunk 4 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Primitives/MessagingEntityType.cs]: "this"? 😉

**Comment by SeanFeldman** [comment]
I think you both meant "incorporate the changes" 😛 
done.

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs [hunk 5]
```diff
@@ -139,7 +139,13 @@ internal AmqpMessageSender(string entityName, MessagingEntityType entityType, Se
                     null);
             request.Map[ManagementConstants.Properties.SequenceNumbers] = new[] { sequenceNumber };
 
-            await this.ExecuteRequestResponseAsync(request);
+            var response = await this.ExecuteRequestResponseAsync(request);
+
+            // TODO: no idea what code and exception should be thrown, but the should be
+            if (response.StatusCode != AmqpResponseStatusCode.OK)
+            {
+                // throw
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs]: @SeanFeldman , can you please merge with the latest pack and then throw 
throw response.ToMessagingContractException();

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs]: >  latest pack

@vinaysurya I'm not up to speed with the team\`s jargon - you mean `dev` branch?

**Comment by SeanFeldman** [comment]
@vinaysurya I had to merge conflicts and since no idea if the tests are green or not as it takes forever to run them :( Let me know if something got screwed in the process. Thanks.

**Comment by vinaysurya** [comment]
@SeanFeldman, looks good to me. Not sure where the rest of the comments got lost, but regarding your comment about unused constants in LockTokenName, PrefilteredHeaderName etc in AmqpMessageConverter.cs, yes lets take them off too

**Comment by SeanFeldman** [comment]
rebased on the latest `dev` and removed unused constants from `AmqpMessageConverter.cs`.
@vinaysurya ready for your review/merge (getting tired of rebasing and resolving conflicts 😺 )

**Comment by SeanFeldman** [comment]
Feeli clean! How about you?
image

**Pull request merged**